### PR TITLE
Replace RouteKey with string

### PR DESCRIPTION
**Pull request by Keksoj**
HTTP and HTTPS Frontends are stored in `ConfigState` using `RouteKey` as a key in BTreeMaps. RouteKey is a sum-up of what makes a frontend unique: address, hostname, path rule. Thus we avoid collisions in the BTreeMap.

```rust
struct ConfigState {
     // ...
     pub http_fronts: BTreeMap<RouteKey, HttpFrontend>
}
```

```rust
pub struct RouteKey {
    pub address: String,
    hostname: String,
    pub path_rule: PathRule,
    pub method: Option<String>,
}
```

Using a struct as key in a map is difficult to translate and serialize in protobuf though, so instead of using RouteKey I rewrote the maps to use a String as key, and this string is the exact same thing RouteKey was serialized to.

Since they are no more uses for RouteKey, I removed it.

Also, ConfigState has a `https_fronts` field but it is never populated. The HTTPS frontends are stored in `http_fronts`. So I removed the `https_fronts` field altogether, and then decided to populate it instead.

### CODE REVIEW

File: command/src/request.rs [hunk 1]
```diff
@@ -350,6 +345,27 @@ impl RequestHttpFrontend {
             tags: self.tags,
         })
     }
+
+    /// Used to create a unique summary of the frontend, used as a key in maps
+    pub fn to_string(&self) -> String {
```
Comment (FlorentinDUBOIS) [inline command/src/request.rs:350]: Could you implements `Display` instead ?

File: command/src/state.rs [hunk 2]
```diff
@@ -52,10 +50,9 @@ pub struct ConfigState {
     pub http_listeners: HashMap<SocketAddr, (HttpListenerConfig, bool)>,
     pub https_listeners: HashMap<SocketAddr, (HttpsListenerConfig, bool)>,
     pub tcp_listeners: HashMap<SocketAddr, (TcpListenerConfig, bool)>,
-    /// indexed by (address, hostname, path)
-    pub http_fronts: BTreeMap<RouteKey, HttpFrontend>,
-    /// indexed by (address, hostname, path)
-    pub https_fronts: BTreeMap<RouteKey, HttpFrontend>,
+    /// HTTP and HTTPS frontends, indexed by a summary of each front's address;hostname;path, for uniqueness.
```
Comment (FlorentinDUBOIS) [inline command/src/state.rs:53]: How do you discriminate frontends with tls from whose that they do not have tls enabled?

File: command/src/state.rs [hunk 3]
```diff
@@ -864,11 +857,11 @@ impl ConfigState {
             }
         }
 
-        let mut my_http_fronts: HashSet<(&RouteKey, &HttpFrontend)> = HashSet::new();
+        let mut my_http_fronts: HashSet<(&String, &HttpFrontend)> = HashSet::new();
```
Comment (FlorentinDUBOIS) [inline command/src/state.rs:860]: Should be `&str`

File: command/src/state.rs [hunk 4]
```diff
@@ -864,11 +857,11 @@ impl ConfigState {
...
         for (route, front) in self.http_fronts.iter() {
             my_http_fronts.insert((route, front));
         }
-        let mut their_http_fronts: HashSet<(&RouteKey, &HttpFrontend)> = HashSet::new();
+        let mut their_http_fronts: HashSet<(&String, &HttpFrontend)> = HashSet::new();
```
Comment (FlorentinDUBOIS) [inline command/src/state.rs:864]: Should be `&str`

**Review by FlorentinDUBOIS — CHANGES_REQUESTED** [review]

File: command/src/state.rs (hunk 2 above)
Comment (Keksoj) [reply command/src/state.rs:53]: - #909

File: command/src/state.rs (hunk 3 above)
Comment (Keksoj) [reply command/src/state.rs]: done!

File: command/src/state.rs (hunk 4 above)
Comment (Keksoj) [reply command/src/state.rs]: done!

**Review by FlorentinDUBOIS — APPROVED** [review]
LGTM

**Pull request merged**